pkg/threescale/connectors/backend/cached_backend.go

```diff
@@ -41,48 +50,24 @@ type CachedBackend struct {
 	hierarchyTree hierarchyTree
 }
 
-type LocalCache struct {
-	ds cmap.ConcurrentMap
-}
-
 // map to represent parent --> children metric relationship
 type metricParentToChildren map[string][]string
 
 // hierarchyTree - represents the relationship between service metrics and their children
 type hierarchyTree map[string]metricParentToChildren
 
 // NewCachedBackend returns a pointer to a CachedBackend with a default LocalCache if no custom implementation has been provided
-func NewCachedBackend(cache Cacheable, rfn metrics.ReportMetricsFn) *CachedBackend {
+// and a channel which can be closed to stop the background process from reporting.
+func NewCachedBackend(cache Cacheable, rfn metrics.ReportMetricsFn) (*CachedBackend, chan struct{}) {
```

Comment:
Well, i thought we wouldn't need to carry it anywhere, it would be up to the caller to close it when they want to stop the worker but I see your point.

So you are suggesting to add a field to the `CachedBackend` to hold the channel and let the user pass it in here as param?

Reply:
well, if it is included in `CachedBackend` then the param disappears, leaving this as it was.